src/main/java/teammates/logic/api/AccessControlUtil.java

```diff
@@ -1,5 +1,239 @@
+            List<String> sectionStudentsEmails, List<String> teamsInSection) {
+        // if recipient type is not a student or team, this comment is not directed at a recipient in the
+        // same section as the student
+        if (!CommentParticipantType.PERSON.equals(comment.recipientType)
+                && !CommentParticipantType.TEAM.equals(comment.recipientType)) {
+            return false;
+        }
+        
+        // check if student is in same section
+        if (CommentParticipantType.PERSON.equals(comment.recipientType)
+                && isCommentRecipientPresentInList(comment, sectionStudentsEmails)) {
+            return true;
+            
+        // check if team is in same section
+        } else if (isCommentRecipientPresentInList(comment, teamsInSection)) {
+            return true;
+        }
+        
+        return false;
+    }
+    
+    private boolean isStudentCourseRecipientOfComment(CommentAttributes comment, StudentAttributes student) {
+        return CommentParticipantType.COURSE.equals(comment.recipientType) && comment.courseId.equals(student.course);
+    }
+    
+    private boolean isStudentInSameCourseAsRecipientOfComment(CommentAttributes comment, StudentAttributes student) {
+        return comment.courseId.equals(student.course);
+    }
+    
+    /**
+     * Checks if the recipient of the comment is found in the provided list of emails/names.
+     * Preconditions: comment is not null.
+     */
+    private boolean isCommentRecipientPresentInList(CommentAttributes comment, List<String> candidateRecipients) {
+        if (!comment.recipients.isEmpty() && candidateRecipients != null) {
+            String commentRecipient = comment.recipients.iterator().next();
+            return candidateRecipients.contains(commentRecipient);
+        }
+        return false;
+    }
+    
+    /**
+     * Returns a {@code List} of boolean values representing the permissions a comment has for a specific participant type.
+     * Preconditions: No parameter is null.
+     */
+    private List<Boolean> getPermissionsForComment(
+            CommentAttributes comment, CommentParticipantType participantType, boolean isRecipientOfComment) {
+        List<Boolean> permissions = new ArrayList<Boolean>();
+        for (int i = 0; i < COMMENT_PERMISSIONS_TOTAL_SIZE; i++) {
+            permissions.add(false);
+        }
+        
+        permissions.set(COMMENT_PERMISSIONS_COMMENT_IS_DISPLAYED_INDEX, comment.showCommentTo.contains(participantType));
+        permissions.set(COMMENT_PERMISSIONS_GIVER_IS_DISPLAYED_INDEX, comment.showGiverNameTo.contains(participantType));
+        // if the person is the recipient of the comment, do not hide the recipient name
+        permissions.set(COMMENT_PERMISSIONS_RECIPIENT_IS_DISPLAYED_INDEX,
+                        isRecipientOfComment || comment.showRecipientNameTo.contains(participantType));
+        return permissions;
```

Comment:
This might already be in your consideration for the next few PRs, but in case it's not: consider using an AccessControl DTO rather than a boolean list with magic numbers (it's not so "magic" after being extracted to constants, but still quite fragile).